utils/cautils/certificate_flow.go

```diff
@@ -286,10 +286,25 @@ func (f *CertificateFlow) CreateSignRequest(ctx *cli.Context, tok, subject strin
 			dnsNames, ips, emails, uris = splitSANs(defaultSANs)
 		}
 	case token.OIDC:
-		if jwt.Payload.Email != "" {
-			emails = append(emails, jwt.Payload.Email)
+		// If no sans are given using the --san flag, and the subject argument
+		// matches the email then CN=token.sub SANs=email, token.iss#token.sub
+		//
+		// If no sans are given and the subject argument does not match the
+		// email then CN=subject SANs=splitSANs(subject)
+		//
+		// If sans are provided CN=subject SANs=splitSANs(sans)
+		if len(sans) == 0 {
+			if jwt.Payload.Email != "" && strings.EqualFold(subject, jwt.Payload.Email) {
+				subject = jwt.Payload.Subject
+				emails = append(emails, jwt.Payload.Email)
+				if iss, err := url.Parse(jwt.Payload.Issuer); err == nil && iss.Scheme != "" {
+					iss.Fragment = jwt.Payload.Subject
+					uris = append(uris, iss)
+				}
+			} else {
+				dnsNames, ips, emails, uris = splitSANs([]string{subject})
```

Comment:
what about the SANs in jwt.Payload.SANs?

Reply:
https://github.com/smallstep/cli/pull/551/files#diff-d45e314b2cc37cd26b0c92007c7a6dabc13a7006602b51bf9f5e9b4729b8c1ebR252

Reply:
I've looked at this again, and if you add the "sans" claim the logic thinks that is a JWT provisioner, so it won't go here. So right now we don't support the custom claim "sans" in an OIDC token.

Reply:
At the very least let's document that invariant. But I'd rather explicitly set dnsNames, ips, emails, uris to empty lists. If that invariant ever changes, and we're using append to existing lists then the behavior will silently change.

Reply:
Fixed in 245ddf1